core/src/main/java/org/mule/runtime/core/el/DefaultExpressionManager.java

```diff
@@ -0,0 +1,249 @@
+    DataType sourceType = result.getDataType();
+    try {
+      Object trans = muleContext.getRegistry().lookupTransformer(sourceType, outputType).transform(result.getValue());
+      return new DefaultTypedValue(trans, outputType);
+    } catch (TransformerException e) {
+      throw new ExpressionRuntimeException(createStaticMessage(
+                                                               format("Failed to apply implicit transformation from type %s to %s",
+                                                                      sourceType, outputType),
+                                                               e));
+    }
+  }
+
+  @Override
+  public void enrich(String expression, Event event, Event.Builder eventBuilder, FlowConstruct flowConstruct, Object object) {
+    selectExpressionLanguage(expression).enrich(expression, event, eventBuilder, flowConstruct, object);
+  }
+
+  @Override
+  public void enrich(String expression, Event event, Event.Builder eventBuilder, FlowConstruct flowConstruct, TypedValue value) {
+    selectExpressionLanguage(expression).enrich(expression, event, eventBuilder, flowConstruct, value);
+  }
+
+  @Override
+  public boolean evaluateBoolean(String expression, Event event, FlowConstruct flowConstruct)
+      throws ExpressionRuntimeException {
+    return evaluateBoolean(expression, event, flowConstruct, false, false);
+  }
+
+  @Override
+  public boolean evaluateBoolean(String expression, Event event, FlowConstruct flowConstruct, boolean nullReturnsTrue,
+                                 boolean nonBooleanReturnsTrue)
+      throws ExpressionRuntimeException {
+    return resolveBoolean(evaluate(expression, event, flowConstruct).getValue(), nullReturnsTrue, nonBooleanReturnsTrue,
+                          expression);
+  }
+
+  protected boolean resolveBoolean(Object result, boolean nullReturnsTrue, boolean nonBooleanReturnsTrue, String expression) {
+    if (result == null) {
+      return nullReturnsTrue;
+    } else {
+      Object value = result;
+      if (value instanceof Boolean) {
+        return (Boolean) value;
+      } else if (value instanceof String) {
+        if (value.toString().toLowerCase().equalsIgnoreCase("false")) {
+          return false;
+        } else if (result.toString().toLowerCase().equalsIgnoreCase("true")) {
+          return true;
+        } else {
+          return nonBooleanReturnsTrue;
+        }
+      } else {
+        logger.warn("Expression: " + expression + ", returned an non-boolean result. Returning: " + nonBooleanReturnsTrue);
+        return nonBooleanReturnsTrue;
+      }
+    }
+  }
+
+  @Override
+  public String parse(String expression, Event event, FlowConstruct flowConstruct) throws ExpressionRuntimeException {
+    return parse(expression, event, Event.builder(event), flowConstruct);
+  }
+
+  @Override
+  public String parse(String expression, Event event, Event.Builder eventBuilder, FlowConstruct flowConstruct)
```

Comment:
I understand it depends on the evaluate method but even if we can make it work with DW, we don't want it to work.